util/vendor.py

```diff
@@ -543,6 +573,25 @@ def git_add_commit(paths, commit_msg):
         log.warning("Unable to create commit. Are there no changes?")
 
 
+def define_arg_type(overridable_keys):
+    def parse(arg):
+        """Sanity-check and return a config file override argument"""
+        try:
+            (key, value) = [v.strip() for v in arg.split('=', 2)]
+        except Exception:
+            raise argparse.ArgumentTypeError(
+                'unable to parse {!r}: configuration overrides must be in the form key=value'
+                .format(arg))
+
+        if key not in overridable_keys:
+            raise argparse.ArgumentTypeError(
+                'invalid configuration override: key {!r} cannot be overwritten'
+                .format(key))
+        return (key, value)
+
+    return parse
```

Comment:
I don't think you actually need the closure here, since you only ever call it with the (global, constant) `OVERRIDABLE_DESC_KEYS`. Probably cleaner to unpack it?